pkg/substitution/substitution.go

```diff
@@ -14,27 +15,27 @@ const (
 	cloudNodeManagerName       = "cloud-node-manager"
 )
 
-// setDeploymentImages substitutes controller containers in Deployment with correct image
-func setDeploymentImages(config config.OperatorConfig, d *v1.Deployment) {
-	for i, container := range d.Spec.Template.Spec.Containers {
-		if container.Name != cloudControllerManagerName {
+// setCloudControllerImage substitutes controller containers in provided pod specs with correct image
+func setCloudControllerImage(config config.OperatorConfig, p corev1.PodSpec) corev1.PodSpec {
+	updatedPod := *p.DeepCopy()
+	for i, container := range p.Containers {
+		substitute := ""
+		switch container.Name {
+		case cloudControllerManagerName:
+			substitute = config.ControllerImage
+		case cloudNodeManagerName:
+			substitute = config.CloudNodeImage
+		default:
 			continue
 		}
 
-		klog.Infof("Substituting %q in %q with %s", container.Name, d.Kind, config.ControllerImage)
-		d.Spec.Template.Spec.Containers[i].Image = config.ControllerImage
-	}
-}
-
-func setDaemonSetImage(config config.OperatorConfig, d *v1.DaemonSet) {
-	for i, container := range d.Spec.Template.Spec.Containers {
-		if container.Name != cloudNodeManagerName {
-			continue
+		if substitute != "" {
+			klog.Infof("Substituting %q in with %s", container.Name, substitute)
```

Comment:
Perhaps a bit more explicit what's happening here
```suggestion
			klog.Infof("Substituting container image for container %q with %q", container.Name, substitute)
```